itest/assets_test.go

```diff
@@ -584,31 +584,42 @@ func sendAssetKeySendPayment(t *testing.T, src, dst *HarnessNode, amt uint64,
 
 	srcTapd := newTapClient(t, src)
 
-	// Now that we know the amount we need to send, we'll convert that into
-	// an HTLC tlv, which'll be used as the first hop TLV value.
-	encodeReq := &tchrpc.EncodeCustomRecordsRequest_RouterSendPayment{
-		RouterSendPayment: &tchrpc.RouterSendPaymentData{
-			AssetAmounts: map[string]uint64{
-				hex.EncodeToString(assetID): amt,
-			},
-		},
+	// Read out the custom preimage for the keysend payment.
+	var preimage lntypes.Preimage
+	_, err := rand.Read(preimage[:])
+	require.NoError(t, err)
+
+	hash := preimage.Hash()
+
+	// Set the preimage. If the user supplied a preimage with the data
+	// flag, the preimage that is set here will be overwritten later.
+	customRecords := make(map[uint64][]byte)
+	customRecords[record.KeySendType] = preimage[:]
+
+	sendReq := &routerrpc.SendPaymentRequest{
+		Dest:              dst.PubKey[:],
+		Amt:               btcAmt.UnwrapOr(500),
+		DestCustomRecords: customRecords,
+		PaymentHash:       hash[:],
+		TimeoutSeconds:    3,
 	}
-	encodeResp, err := srcTapd.EncodeCustomRecords(
-		ctxt, &tchrpc.EncodeCustomRecordsRequest{
-			Input: encodeReq,
-		},
-	)
+
+	stream, err := srcTapd.SendPayment(ctxt, &tchrpc.SendPaymentRequest{
+		AssetId:        assetID,
+		AssetAmount:    amt,
+		PaymentRequest: sendReq,
+	})
 	require.NoError(t, err)
 
-	htlcCarrierAmt := btcAmt.UnwrapOr(500)
-	sendKeySendPayment(
-		t, src, dst, btcutil.Amount(htlcCarrierAmt),
-		encodeResp.CustomRecords,
-	)
+	time.Sleep(time.Second)
```

Comment:
Unnecessary sleep? Shouldn't `getAssetPaymentResult` block on the recv?

Reply:
Unfortunately and strangely the tests fail without it... It shouldn't make a difference, but somehow it does. Moved it to a single place and added an explanatory comment.